### ISSUE TITLE
`Model::query()->first()` doesn't return correct type with Dedicated Builders

### ISSUE THREAD

**Issue by MatanYadaev** [barryvdh/laravel-ide-helper#1136]
### Versions:
- ide-helper Version: 2.8.2
- Laravel Version: 8.20.1
- PHP Version: 8.0.0

### Description:

I'm [Dedicated Builders](https://www.google.com/search?sxsrf=ALeKk03ZG2qxEaqIaKjj6pb-SgMk6Cjyvg%3A1609697150658&ei=fgfyX5DWJ4iFhbIPk8SxOA&q=dedicated+builders+laravel&oq=dedicated+builders+laravel&gs_lcp=CgZwc3ktYWIQAzIFCCEQoAE6BwgjEMkDECc6BggAEBYQHjoJCAAQyQMQFhAeUPkMWKgYYNoYaABwAHgBgAGjA4gBoAqSAQkwLjEuMy4wLjGYAQCgAQGqAQdnd3Mtd2l6wAEB&sclient=psy-ab&ved=0ahUKEwiQjfGWrYDuAhWIQkEAHRNiDAcQ4dUDCA0&uact=5) in my project, and as a result of it, `User::query()->first()` phpDoc return `Model` instead of `User`.

Code example:
```php
/**
 * @method static UserBuilder|User query()
 */
class User extends Authenticatable
{
  // ....
 public function newEloquentBuilder($query)
  {
    return new UserBuilder($query);
  }
  // ...
}
```

```php
/** WORKS AS EXPECTED */
// `$user` is `\Illuminate\Database\Eloquent\Model|static` (static = User)
$user = User::first();

/** DOESN'T WORK AS EXPECTED */
// `$userBuilder` is `App\Builders\UserBuilder`
$userBuilder = User::query();
// `$user` is `\Illuminate\Database\Eloquent\Model|static` (static = UserBuilder)
$user = $userBuilder->first();
dump($user->first_name); // `first_name` doesn't get auto-complete
```

To summarize, `laravel-ide-helper` adds the `@return Model|static` phpDoc, it's great when it's under the `User`, but doesn't work well when it's under `UserBuilder`.
